staging/src/k8s.io/apiserver/pkg/apis/apiserver/validation/validation.go

```diff
@@ -45,32 +45,24 @@ func ValidateAuthenticationConfiguration(c *api.AuthenticationConfiguration) fie
 	root := field.NewPath("jwt")
 	var allErrs field.ErrorList
 
-	// This stricter validation is solely based on what the current implementation supports.
-	// TODO(aramase): when StructuredAuthenticationConfiguration feature gate is added and wired up,
-	// relax this check to allow 0 authenticators. This will allow us to support the case where
-	// API server is initially configured with no authenticators and then authenticators are added
-	// later via dynamic config.
-	if len(c.JWT) == 0 {
-		allErrs = append(allErrs, field.Required(root, fmt.Sprintf(atLeastOneRequiredErrFmt, root)))
-		return allErrs
-	}
+	// We allow 0 authenticators in the authentication configuration.
+	// This allows us to support scenarios where the API server is initially set up without
+	// any authenticators and then authenticators are added later via dynamic config.
 
-	// This stricter validation is because the --oidc-* flag option is singular.
-	// TODO(aramase): when StructuredAuthenticationConfiguration feature gate is added and wired up,
-	// remove the 1 authenticator limit check and add set the limit to 64.
-	if len(c.JWT) > 1 {
-		allErrs = append(allErrs, field.TooMany(root, len(c.JWT), 1))
+	if len(c.JWT) > 64 {
```

Comment:
See https://github.com/kubernetes/kubernetes/issues/122809#issuecomment-1894818219 for more details.

> Shouldn't that number rather be a documented constant such as `MaximumAuthenticatorNumber` so it may be more easily changed ?

I can do this.

Reply:
Not sure if using a documented constant here matters much since the ideal long term is to remove the limit altogether (i.e. before GA seems best).  The number itself is also somewhat arbitrary since we didn't want to impact any normal use case, and just want to avoid users getting stuck in a bad edge case.

Reply:
I see your point, then I think it would be good to keep comment or maybe a todo in order to not lose the link to the original decision and wonder later on why that 64.

A comment about it in the commit message would also be of value.